src/main/java/seedu/tassist/logic/parser/MarkAttendanceCommandParser.java

```diff
@@ -25,22 +28,40 @@ public class MarkAttendanceCommandParser implements Parser<MarkAttendanceCommand
      */
     public MarkAttendanceCommand parse(String args) throws ParseException {
         requireNonNull(args);
-        ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_INDEX, PREFIX_WEEK);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(
+                args, PREFIX_INDEX, PREFIX_WEEK, PREFIX_MARK_UNATTENDED, PREFIX_MARK_ON_MC);
 
         Index index;
         int week;
+        boolean isUnattended = false;
+        boolean isOnMc = false;
 
         try {
             index = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).orElse(""));
             week = ParserUtil.parseWeek(argMultimap.getValue(PREFIX_WEEK).orElse(""));
+            isUnattended = !argMultimap.getValue(PREFIX_MARK_UNATTENDED).isEmpty();
+            isOnMc = !argMultimap.getValue(PREFIX_MARK_ON_MC).isEmpty();
         } catch (IllegalValueException ive) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                             MarkAttendanceCommand.MESSAGE_USAGE), ive
             );
         }
 
-        return new MarkAttendanceCommand(index, week);
+        if (isUnattended && isOnMc) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                            MarkAttendanceCommand.MESSAGE_USAGE)
+            );
+        }
```

Comment:
Break line for logic segregation
```suggestion
        }
        
```